pkg/grpc/exec/exec.go

```diff
@@ -273,14 +273,15 @@ func (msg *MsgCloneEventUnix) Cast(o interface{}) notify.Message {
 func GetProcessExit(event *MsgExitEventUnix) *tetragon.ProcessExit {
 	var tetragonProcess, tetragonParent *tetragon.Process
 
-	process, parent := process.GetParentProcessInternal(event.ProcessKey.Pid, event.ProcessKey.Ktime)
-	if process != nil {
-		tetragonProcess = process.UnsafeGetProcess()
+	proc, parent := process.GetParentProcessInternal(event.ProcessKey.Pid, event.ProcessKey.Ktime)
+	if proc != nil {
+		tetragonProcess = proc.UnsafeGetProcess()
 	} else {
 		tetragonProcess = &tetragon.Process{
 			Pid:       &wrapperspb.UInt32Value{Value: event.ProcessKey.Pid},
 			StartTime: ktime.ToProto(event.ProcessKey.Ktime),
 		}
+		process.UpdateEventProcessTid(tetragonProcess, event.Info.Tid)
```

Comment:
Shouldn't we call UpdateEventProcessTid for other paths as well?

Reply:
For the `GetProcessExit->GetParentProcessInternal()` it should return a process that has the TID set from the exec event or from the clone one. Otherwise if we miss the cache we use the one that BPF recorded. These are the only path here

Reply:
Ok @kkourt , updated and made it simpler that all paths use the `UpdateEventProcessTid()`